src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/NullableConverter.cs

```diff
@@ -70,5 +70,7 @@ internal override void WriteNumberWithCustomHandling(Utf8JsonWriter writer, T? v
                 _converter.WriteNumberWithCustomHandling(writer, value.Value, handling);
             }
         }
+
+        internal override bool IsNull(T? value) => !value.HasValue;
```

Comment:
This does seem a bit strange. I'll do some local prototyping to see why is going on here -- boxing issues with `Nullable<T>` or otherwise. If there's an issue with `Nullable<T>` then we should try to fix it in the runtime.